src/game_battler.cpp

```diff
@@ -458,11 +458,11 @@ void Game_Battler::RemoveBattleStates() {
 	// If death is non-permanent change HP to 1
 	if (IsDead() &&
 		non_permanent(1)) {
-		ChangeHp(1);
+		RemoveState(1);
 	}
 
-	for (size_t i = 0; i < states.size(); ++i) {
-		if (non_permanent(i + 1)) {
+	for (size_t i = 1; i < states.size(); ++i) {
+		if (non_permanent(i + 1) || ReaderUtil::GetElement(Data::states, i + 1)->auto_release_prob == 0) {
```

Comment:
Oh I just remembered something!
The commit explanation is correct (> 0% chance of recovery will heal after battle) but the code is wrong, change == to > xD

Reply:
Fixed